apis/management.cattle.io/v3/multi_cluster_app_types.go

```diff
@@ -13,3 +16,67 @@ type GlobalDNS struct {
 	TTLSeconds   int64  `json:"ttl" norman:"default=300"`
 	ProviderName string `json:"providerName,omitempty" norman:"required"`
 }
+
+type MultiClusterApp struct {
+	types.Namespaced
+
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MultiClusterAppSpec   `json:"spec,omitempty"`
+	Status MultiClusterAppStatus `json:"status,omitempty"`
+}
+
+type MultiClusterAppSpec struct {
+	ReleaseName string             `json:"releaseName,omitempty"`
+	Answers     map[string]string  `json:"answers,omitempty"`
+	HelmChart   HelmChartReference `json:"helmChart,omitempty"`
+	Template    TemplateReference  `json:"template,omitempty"`
+	Targets     []Target           `json:"targets,omitempty" norman:"required"`
+}
+
+type MultiClusterAppStatus struct {
+	HealthState string `json:"healthState,omitempty" norman:"required"`
+}
+
+type HelmChartReference struct {
+	RepositoryURL string `json:"chartRepositoryUrl,omitempty"`
+	Reference     string `json:"chartReference,omitempty"`
+	Version       string `json:"chartVersion,omitempty"`
+}
+
+type TemplateReference struct {
+	TemplateName    string `json:"templateName,omitempty" norman:"required,type=reference[template]"`
+	TemplateVersion string `json:"templateVersion,omitempty" norman:"required,type=reference[templateVersion]"`
+}
+
+type Target struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   TargetSpec   `json:"spec,omitempty" norman:"required"`
+	Status TargetStatus `json:"status,omitempty" norman:"required"`
+}
+
+type TargetSpec struct {
+	ClusterName      string            `json:"clusterName,omitempty" norman:"type=reference[cluster]"`
+	ClusterConfig    ClusterConfig     `json:"clusterConfig,omitempty"`
```

Comment:
ClusterConfig should be of a pointer type as it won't be required when the tool is deployed in rancher